File: hackerrank/10 Days of Statistics/Day 0: Weighted Mean/solution.js
function processData(input) {
    input = input.split('\n');

    const n = parseInt(input[0]);
    const x = input[1].split(' ').map((n) => parseInt(n)).slice(0, n);
    const w = input[2].split(' ').map((n) => parseInt(n)).slice(0, n);

    const x1 = x.reduce((acc, value, index) => acc + value * w[index], 0);
    const w1 = w.reduce((acc, value) => acc + value, 0);

    console.log((x1 / w1).toFixed(1));
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
    processData(_input);
});

File: hackerrank/10 Days of Statistics/Day 0: Weighted Mean/solution.py
from functools import reduce

n = int(input())
x = list(map(int, input().split()))[:n]
w = list(map(int, input().split()))[:n]

x = reduce(lambda acc, value: acc + value[1] * w[value[0]], enumerate(x), 0)
w = reduce(lambda acc, value: acc + value, w, 0)

print(round(x / w, 1))
